pkg/pdp/tasks/watcher_eth.go

```diff
@@ -112,70 +181,228 @@ func (mw *MessageWatcherEth) update() {
 		return
 	}
 
-	// Check if any of the transactions we have assigned are now confirmed
+	if len(txs) == 0 {
+		return
+	}
+
+	log.Debugw("processing pending transactions", "count", len(txs))
+
+	// Create a channel for results
+	resultCh := make(chan *TransactionResult, len(txs))
+
+	// Use errgroup for better error handling and concurrency control
+	g, ctx := errgroup.WithContext(ctx)
+	// limit the number of concurrent executions (a semaphore)
+	g.SetLimit(maxConcurrentChecks)
+
+	// Process transactions concurrently
 	for _, tx := range txs {
 		txHash := common.HexToHash(tx.SignedTxHash)
 
-		receipt, err := mw.api.TransactionReceipt(ctx, txHash)
-		if err != nil {
-			if errors.Is(err, ethereum.NotFound) {
-				// Transaction is still pending
+		g.Go(func() error {
+			result, err := mw.checkTransaction(ctx, txHash, bestBlockNumber)
+			if err != nil {
+				log.Errorf("failed to check transaction %s: %+v", txHash.Hex(), err)
+				// Don't return error - continue processing other transactions
+				return nil
+			}
+
+			if result != nil {
+				select {
+				case resultCh <- result:
+				case <-ctx.Done():
+					return ctx.Err()
+				}
+			}
+
+			return nil
+		})
+	}
+
+	// Process results sequentially to avoid database conflicts
+	var updateCount int
+	wg := sync.WaitGroup{}
```

Comment:
nit: arguably a wait group for just one go routine is overkill (you can just make a done channel for example) 

non blocking